lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -79,6 +79,60 @@
         env: [{name: ANSIBLE_PARAMIKO_PROXY_COMMAND}]
         ini:
           - {key: proxy_command, section: paramiko_connection}
+        vars:
+          - name: ansible_paramiko_proxy_command
+            version_added: '2.15'
+      ssh_args:
+          description: Only used in parsing ProxyCommand for use in this plugin.
+          default: ''
+          ini:
+              - section: 'ssh_connection'
+                key: 'ssh_args'
+          env:
+              - name: ANSIBLE_SSH_ARGS
+          vars:
+              - name: ansible_ssh_args
+                version_added: '2.7'
+          deprecated:
+              why: In favor of the "proxy_command" option.
+              version: "2.18"
+              alternatives: proxy_command
+      ssh_common_args:
+          description: Only used in parsing ProxyCommand for use in this plugin.
+          ini:
+              - section: 'ssh_connection'
+                key: 'ssh_common_args'
+                version_added: '2.7'
+          env:
+              - name: ANSIBLE_SSH_COMMON_ARGS
+                version_added: '2.7'
+          vars:
+              - name: ansible_ssh_common_args
+          cli:
+              - name: ssh_common_args
+          default: ''
+          deprecated:
+              why: In favor of the "proxy_command" option.
+              version: "2.18"
+              alternatives: proxy_command
+      ssh_extra_args:
+          description: Only used in parsing ProxyCommand for use in this plugin.
+          vars:
+              - name: ansible_ssh_extra_args
+          env:
+            - name: ANSIBLE_SSH_EXTRA_ARGS
+              version_added: '2.7'
+          ini:
+            - key: ssh_extra_args
+              section: ssh_connection
+              version_added: '2.7'
+          cli:
+            - name: ssh_extra_args
+          default: ''
+          deprecated:
+              why: In favor of the "proxy_command" option.
+              version: "2.18"
+              alternatives: proxy_command
```

Comment:
we might want also _paramiko_ versions for each var/env var/ini entry that has higher precedence, JIC people use both connections but need different configuration

Reply:
Since there's already a paramiko-specific option, could the priority just be reversed?

```diff
diff --git a/lib/ansible/plugins/connection/paramiko_ssh.py b/lib/ansible/plugins/connection/paramiko_ssh.py
index 82eae5c77e..5410b12b1b 100644
--- a/lib/ansible/plugins/connection/paramiko_ssh.py
+++ b/lib/ansible/plugins/connection/paramiko_ssh.py
@@ -287,7 +287,7 @@ class Connection(ConnectionBase):
             if proxy_command:
                 break
 
-        proxy_command = proxy_command or self.get_option('proxy_command')
+        proxy_command = self.get_option('proxy_command') or proxy_command
```

Reply:
Probably, although that logic existed before this PR, but probably makes sense.